### DISCUSSION TITLE
How to prevent items to be scattered on ground on player's death with API?

### DISCUSSION THREAD

**Discussion by TetraTheta** [GeorgH93/Minepacks#197] [Q&A] [answered]
I have plan to make backpack not scatter its content on player's death when the player has specific item in his inventory (not backpack's inventory). Yes, it would be like Totem of Undying, for backpack inventory save thingy only.

I have looked into [JavaDoc](https://ci.pcgamingfreaks.at/job/Minepacks%20API/javadoc/), but I think javadoc don't have sufficient information of what the event is about, what the method does etc. That makes me to not figure out what event should I listen to.

Is it [`InventoryClearEvent`](https://ci.pcgamingfreaks.at/job/Minepacks%20API/javadoc/at/pcgamingfreaks/Minepacks/Bukkit/API/Events/InventoryClearEvent.html)? What happens before and after the event?

**Comment by GeorgH93** [comment, marked as answer]
I just added the BackpackDropOnDeathEvent with [this build](https://ci.pcgamingfreaks.at/job/Minepacks/280/artifact/a/Minepacks-2.4.4-SNAPSHOT-Release.jar) (will be available in the API starting with version `2.4.4-SNAPSHOT`). You can just cancel it to prevent the backpack form dropping.

**Reply by TetraTheta** [reply]
Thank you so much for fast response(both answer and code)!
